qemu/tests/hugepage_reset.py

```diff
@@ -39,6 +40,46 @@ def set_hugepage():
         except ValueError as err:
             test.cancel(err)
 
+    def run_stress():
+        def heavyload_install():
+            if session.cmd_status(test_install_cmd) != 0:
+                logging.warning("Could not find installed heavyload in guest, "
+                                "will install it via winutils.iso ")
+                winutil_drive = utils_misc.get_winutils_vol(session)
+                if not winutil_drive:
+                    test.cancel("WIN_UTILS CDROM not found.")
+                install_cmd = params["install_cmd"] % winutil_drive
+                session.cmd(install_cmd)
+
+        logging.info('Loading stress on guest.')
+        stress_duration = params.get("stress_duration", 60)
+        if params["os_type"] == "linux":
+            params['stress_args'] = '--vm %s --vm-bytes 256M --timeout %s' % (
+                    mem // 512, stress_duration)
+            stress = utils_test.VMStress(vm, 'stress', params)
+            stress.load_stress_tool()
+            time.sleep(stress_duration)
+            stress.unload_stress()
```

Comment:
It looks a little risky,  load_stress_tools can only ensure the stress tool start successfully,  but no checking to make sure stress doesn't exist abnormally during the  `stress_duration`.  IMO, `time.sleep(stress_duration)` is too crude.

How about the following change, to ensure stress doesn't exist before the `stress_duration` end.
```
stress.load_stress_tool()
if utils_misc.wait_for(lambda: not stress.app_running, stress_duration):
    test.error(****)
else:
    stress.unload_stress()     
```

Reply:
In load_stress_tool, it doesn't check app_running immediately after launch stress.
```
if not utils_misc.wait_for(self.app_running, self.stress_wait_for_timeout, first=2.0,
                                   text="wait for stress app to start",
                                   step=1.0):
            raise exceptions.TestError("Stress app does not "
                                       "running as expected")
```
It's not reliable to check if stress is running in 'stress_duration' time.

Reply:
Besides, as long as stress is launched, I guess we can trust it to be running before timeout, unless it's been killed which doesn't probably happen in this case.

Reply:
> Besides, as long as stress is launched, I guess we can trust it to be running before timeout, unless it's been killed which doesn't probably happen in this case.

Recently there are some issue with netperf, the reason is netperf exist before the specified time, this is not what we want, that is one reason that i raise this issue.

Reply:
As the case I knew, it's because vt has some issue in checking if netperf is running. Sometimes, netperf is not started normally but vt determine it's running. So the running time is not right.

Reply:
> In load_stress_tool, it doesn't check app_running immediately after launch stress.
> 
> ```
> if not utils_misc.wait_for(self.app_running, self.stress_wait_for_timeout, first=2.0,
>                                    text="wait for stress app to start",
>                                    step=1.0):
>             raise exceptions.TestError("Stress app does not "
>                                        "running as expected")
> ```
> 
> It's not reliable to check if stress is running in 'stress_duration' time.

Yes, that is.  In fact, i have thought about this too and thought about if use `stress_duration - ***`, but the value to reduce is not clear, i do not have good idea about it now.   The gap should not large than 3 second.   How do you think about reduce 3 ?

Reply:
> As the case I knew, it's because vt has some issue in checking if netperf is running. Sometimes, netperf is not started normally but vt determine it's running. So the running time is not right.

Yeap, that is the reason for netperf. And the checking i want to add is to avoid such incorrect implementation in upper layer.

But if you think no risk for your test, i am ok.

Reply:
Thanks. Do you have more comments?

Reply:
No, thanks!